pycytominer/normalize.py

```diff
@@ -161,15 +160,34 @@ def normalize(
         # Subset to only the features measured in the sample query
         fitted_scaler = scaler.fit(profiles.query(samples).loc[:, features])
 
-    # Scale the feature dataframe
+    fitted_scaled = fitted_scaler.transform(feature_df)
+
+    columns = fitted_scaler.columns if method == "spherize" else feature_df.columns
+
     feature_df = pd.DataFrame(
-        fitted_scaler.transform(feature_df),
-        columns=feature_df.columns,
+        fitted_scaled,
+        columns=columns,
         index=feature_df.index,
     )
 
     normalized = meta_df.merge(feature_df, left_index=True, right_index=True)
 
+    if feature_df.shape != profiles.loc[:, features].shape:
+        error_detail = (
+            "Feature dataframe shape does not match original dataframe shape."
+        )
+        context = f"the `{method}` method in `pycytominer.normalize`"
+        raise ValueError(util.create_bug_report_message(error_detail, context))
+
+    if (normalized.shape[0] != profiles.shape[0]) or (
+        normalized.shape[1] != len(features) + len(meta_features)
+    ):
+        error_detail = (
+            "Normalized dataframe shape does not match original dataframe shape."
```

Comment:
I have changed it to `The number of rows and columns in the normalized dataframe does not match the original dataframe` because it needn't be cells (could be well, for example), and we also check the number of columns (not just rows)